tests/test_internals/test_si.py

```diff
@@ -576,5 +585,173 @@ def test_group_set_custom_description(self):
+
+    def test_common_property_extension(self):
+        """Add a property for a first recording that is then extended by a second recording."""
+        self.recording_1.set_property(key="common_property", values=["value_1"] * self.num_channels)
+        self.recording_2.set_property(key="common_property", values=["value_2"] * self.num_channels)
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
+
+        actual_properties_in_electrodes_table = list(self.nwbfile.electrodes["common_property"].data)
+        expected_properties_in_electrodes_table = ["value_1", "value_1", "value_1", "value_1", "value_2", "value_2"]
+        self.assertListEqual(actual_properties_in_electrodes_table, expected_properties_in_electrodes_table)
+
+    def test_new_property_addition(self):
+        """Add a property only available in a second recording."""
+        self.recording_2.set_property(key="added_property", values=["added_value"] * self.num_channels)
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
+
+        actual_properties_in_electrodes_table = list(self.nwbfile.electrodes["added_property"].data)
+        expected_properties_in_electrodes_table = ["", "", "added_value", "added_value", "added_value", "added_value"]
+        self.assertListEqual(actual_properties_in_electrodes_table, expected_properties_in_electrodes_table)
+
+    def test_manual_row_adition_before_add_electrodes_function(self):
+        """Add some rows to the electrode tables before using the add_electrodes function"""
+        values_dic = self.defaults
+
+        values_dic.update(id=123)
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(id=124)
+        self.nwbfile.add_electrode(**values_dic)
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+
+        expected_ids = [123, 124, 2, 3, 4, 5]
+        expected_names = ["123", "124", "a", "b", "c", "d"]
+        self.assertListEqual(list(self.nwbfile.electrodes.id.data), expected_ids)
+        self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), expected_names)
+
+    def test_manual_row_adition_after_add_electrodes_function(self):
+        """Add some rows to the electrode table after using the add_electrodes function"""
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+
+        values_dic = self.defaults
+
+        # Previous properties
+        values_dic.update(rel_x=0.0, rel_y=0.0, id=123, channel_name=str(123))
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(rel_x=0.0, rel_y=0.0, id=124, channel_name=str(124))
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(rel_x=0.0, rel_y=0.0, id=None, channel_name="6")  # automatic ID set
```

Comment:
Note here that the if we have use our `add_electrodes` functio before, the user has to provide the `channel_name` afterwards.

Reply:
Yep, and it will thankfully warn them they need to do so